### PR TITLE
Rename Prisoner Offender Search to Prisoner Search in docs and Docker

### PR DESCRIPTION
**Pull request by chubberlisk**
## Context

The upstream API we use, Prisoner Offender Search, recently renamed to Prisoner Search.

## Changes proposed in this PR

Rename Prisoner Offender Search in our documentation and in Docker.

We decided against making any code changes as it would mean we'd have to make a multitude of changes across a number of different files at once.

### CODE REVIEW

**Review by CslBing — APPROVED** [review]
Looks good

File: docs/diagrams/container.puml [hunk 1]
```diff
@@ -21,10 +21,10 @@ Boundary(aws_cloudplatform, "Cloud Platform", "A Kubernetes cluster using Amazon
     Boundary(upstream, "Legacy Upstream Systems", "Various HMPPS upstream systems, some of these will eventually be replaced"){
         Container(prison_api, "Prison API", "Spring Boot Java API", "API over NOMIS DB used by Digital Prison team applications and services", $link="https://github.com/ministryofjustice/prison-api")
         Container(probation_offender_search, "Probation Offender Search", "Springoot Kotlin API", "API over the nDelius offender data held in Elasticsearch", $link="https://probation-offender-search-dev.hmpps.service.justice.gov.uk/swagger-ui/index.html")
-        Container(prison_offender_search, "Prisoner Offender Search", "Spring Boot Kotlin API", "API over the NOMIS prisoner data held in Elasticsearch", $link="https://github.com/ministryofjustice/prisoner-offender-search")
+        Container(prisoner_search, "Prisoner Search", "Spring Boot Kotlin API", "API over the NOMIS prisoner data held in Elasticsearch", $link="https://github.com/ministryofjustice/hmpps-prisoner-search")
```
Comment (petergphillips) [inline docs/diagrams/container.puml:24]: note that the data these days is held in OpenSearch instead of elastic search

Comment (chubberlisk) [reply docs/diagrams/container.puml:24]: Ahhhhh, thanks Peter for this spot!

File: src/main/resources/application-local-docker.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ services:
   probation-offender-search:
     base-url: http://probation-offender-search:4010
   prisoner-offender-search:
```
[the remark this answers is not in the document]
Comment (chubberlisk) [reply src/main/resources/application-local-docker.yml:4]: Ahhhhhh, this one though was done on purpose as renaming that would propagate a number of changes.

File: docs/diagrams/container.puml (hunk 1 above)
Comment (chubberlisk) [reply docs/diagrams/container.puml:24]: PR created to fix dis: https://github.com/ministryofjustice/hmpps-integration-api/pull/280.

**Pull request merged**